R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
 #' @export
-#' @import tidyverse DBI RMariaDB
+#' @import DBI RMariaDB
+#' @depend tidyverse
 #'
 #' @examples
-#' #basic usage
-#' my_segments = get_cn_segments(region="chr8:128,723,128-128,774,067")
-#' #specifying chromosome, start and end individually
-#' my_segments = get_cn_segments(chromosome="8",qstart=128723128,qend=128774067)
-#' #Asking for chromosome names to have a chr prefix (default is un-prefixed)
-#' prefixed_segments = get_cn_segments(chromosome ="12",qstart = 122456912, qend = 122464036, with_chr_prefix = TRUE)
-#'
-get_cn_segments = function(chromosome = "",
+#' #Example using chromosome, qstart and qend parameters:
+#' segments_region_grch37 = get_cn_segments(chromosome = "chr8",
+#'                                          qstart = 128723128,
+#'                                          qend = 128774067,
+#'                                          projection = "grch37", 
+#'                                          this_seq_type = "genome", 
+#'                                          remove_chr_prefix = FALSE, 
+#'                                          from_flatfile = TRUE, 
+#'                                          streamlined = FALSE) 
+#'                                     
+#' #Example using the regions parameter:
+#' segments_region_hg38 = get_cn_segments(region = "chr8:128,723,128-128,774,067",
+#'                                        projection = "hg38", 
+#'                                        this_seq_type = "genome", 
+#'                                        remove_chr_prefix = TRUE, 
+#'                                        from_flatfile = TRUE, 
+#'                                        streamlined = FALSE) 
+#'
+#'
+get_cn_segments = function(region,
+                           chromosome,
                            qstart,
                            qend,
-                           region,
-                           with_chr_prefix = FALSE,
+                           projection = "grch37",
+                           this_seq_type = "genome",
+                           remove_chr_prefix = FALSE,
                            streamlined = FALSE,
-                           from_flatfile = FALSE){
+                           from_flatfile = TRUE){
+  
+  #checks
   remote_session = check_remote_configuration()
-  db = config::get("database_name")
-  table_name = config::get("results_tables")$copy_number
-  table_name_unmatched = config::get("results_tables")$copy_number_unmatched
+  
+  #check seq type and return a message if anything besides "genome" is called. To be updated once capture samples have been processed through CNV protocols.
+  if(this_seq_type!="genome"){
+    stop("Currently, only genome samples are available for this function. Please select a valid seq type (i.e genome). Compatibility for capture samples will be added soon...")
+  }
+  
+  #get wildcards from this_seq_type (lazy)
+  seq_type = this_seq_type
```

Comment:
I would think this might not be needed here to keep the function simple, and `this_seq_type` can be used as is

Reply:
Creating a variable with the name that matches the wildcard is necessary when we use glue. Don't change this unless we're certain glue isn't being used later

Reply:
Precisely, that is indeed why that variable was created. I will not implement the edit suggested by Kostia.